tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_scanner.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <kernel.h>
+#include "mesh_test.h"
+#include "mesh/net.h"
+#include "mesh/adv.h"
+#include "mesh/mesh.h"
+#include "mesh/foundation.h"
+
+#define LOG_MODULE_NAME test_scanner
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#define WAIT_TIME 60 /*seconds*/
+#define ADV_INT_FAST_MS 20
+#define ADV_DURATION 120
+
+extern enum bst_result_t bst_result;
+
+static const struct bt_mesh_test_cfg tx_cfg = {
+	.addr = 0x0001,
+	.dev_key = { 0x01 },
+};
+static const struct bt_mesh_test_cfg rx_cfg = {
+	.addr = 0x0002,
+	.dev_key = { 0x02 },
+};
+
+static void test_tx_init(void)
+{
+	bt_mesh_test_cfg_set(&tx_cfg, WAIT_TIME);
+}
+
+static void test_rx_init(void)
+{
+	bt_mesh_test_cfg_set(&rx_cfg, WAIT_TIME);
+}
+
+/* Setup the tx device by enabling Bluetooth, but no scanner is needed. */
+static void test_tx_device_setup(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Bluetooth init failed (err %d)", err);
+		return;
+	}
+
+	LOG_INF("Bluetooth initialized");
+}
+
+/** Bypassing setting up transmission, and will try to send the raw data that is
+ *  provided to the function.
+ */
+static void test_tx_send_ad_type_msg(uint8_t type, uint8_t *data, uint8_t len)
+{
+	struct bt_le_adv_param param = {};
+	uint16_t duration, adv_int;
+	struct bt_data ad;
+	int err;
+
+	adv_int = ADV_INT_FAST_MS;
+	duration = ADV_DURATION;
+
+	ad.type = type;
+	ad.data_len = len;
+	ad.data = data;
+
+	param.id = BT_ID_DEFAULT;
+	param.interval_min = BT_MESH_ADV_SCAN_UNIT(adv_int);
+	param.interval_max = param.interval_min;
+
+	LOG_DBG("ad.type %u len %u", ad.type, ad.data_len);
+	LOG_HEXDUMP_DBG(ad.data, ad.data_len, "ad.data");
+
+	uint64_t st = k_uptime_get();
+
+	err = bt_le_adv_start(&param, &ad, 1, NULL, 0);
+	if (err) {
+		FAIL("Advertising failed: err %d", err);
+		return;
+	}
+	LOG_DBG("Advertising started. Sleeping %u ms", duration);
+
+	k_sleep(K_MSEC(duration));
+	err = bt_le_adv_stop();
+	if (err) {
+		FAIL("Stopping advertising failed: err %d", err);
+		return;
+	}
+	LOG_DBG("Advertising stopped (%u ms)", (uint32_t) k_uptime_delta(&st));
+}
+
+/** Test sending message with not supported ad type for mesh.
+ *
+ * First message send message with an ad data type not supported by mesh, and
+ * verify that the receiver is disregarding message. Then send the same message
+ * with correct ad type to verify that the message is received.
+ */
+static void test_tx_invalid_ad_type(void)
+{
+	test_tx_device_setup();
+
+	/* Create a valid message */
+	uint8_t data[28] = {
+		0x0d, 0x10, 0xca, 0x54, 0xd0,
+		0x00, 0x24, 0x00, 0xaa, 0x8c,
+		0xcc, 0x6b, 0x6a, 0xc8, 0x51,
+		0x69, 0x16, 0x4d, 0xf6, 0x9b,
+		0xce, 0xbd, 0xc7, 0xa3, 0xf0,
+		0x28, 0xdf, 0xae
+	};
```

Comment:
Perhaps this could be a `static const uint8_t valid_message[28]`  at the top of the file? It'd also be nice with some annotations, like in the beacon sample: https://github.com/zephyrproject-rtos/zephyr/blob/57e7fba288cb0b48cc9dbc1190caacd99d548bd7/samples/bluetooth/beacon/src/main.c#L28-L35

Reply:
I moved the message to the top of the file. But I did not add any annotations as I do not see that we can change the message in any way. Or I have to decode the message and I do not know if that has any meaning as this is only a test.